client/lib/formatting/prevent-widows.js

```diff
@@ -6,35 +11,70 @@
  * @returns {string}             the widow-prevented string
  */
 export function preventWidows( text, wordsToKeep = 2 ) {
-	if ( typeof text !== 'string' ) {
-		if ( Array.isArray( text ) ) {
-			// Handle strings with interpolated components by only acting on the last element.
-			if ( typeof text[ text.length - 1 ] === 'string' ) {
-				const endingText = text.pop();
-				const startingWhitespace = endingText.match( /^\s+/ );
-				// The whitespace between component and text would be stripped by preventWidows.
-				startingWhitespace && text.push( startingWhitespace[ 0 ] );
-				text.push( preventWidows( endingText, wordsToKeep ) );
-			}
-		}
-		return text;
-	}
+	return preventWidowsInPart( text, Math.max( 1, wordsToKeep - 1 ) ).part;
+}
 
-	text = text && text.trim();
-	if ( ! text ) {
-		return text;
-	}
+const reverseSpaceRegex = /\s+(\S*)$/;
 
-	const words = text.match( /\S+/g );
-	if ( ! words || 1 === words.length ) {
-		return text;
+/**
+ * The hulper function to preventWidows that calls itself rucursively searching for spaces to substitute with
```

Comment:
hulper? should that be helper?!

Reply:
also rucrsively? did you change your keyboard layout?!

Reply:
No, not recently! Monday typing.